### PR TITLE
Patch search.tasks instead of search.indexing_api

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Part of #2705

#### What's this PR do?
Patches search.tasks instead of search.indexing_api because patches should patch the module where it's invoked rather than the one where it's defined. This worked fine before because `search.indexing_api` functions usually just called other functions, but in a future PR this will not work because the percolate indexing work uses a function in `search.api`

#### How should this be manually tested?
This should only affect tests so no manual review is necessary

### CODE REVIEW

File: search/base.py [hunk 1]
```diff
@@ -44,9 +44,9 @@ class MockedESTestCase(TestCase):
     @classmethod
     def setUpClass(cls):
         cls.patchers = []
-        for name, val in indexing_api.__dict__.items():
-            if callable(val):
-                cls.patchers.append(patch('search.indexing_api.{0}'.format(name), autospec=True))
+        for name, val in tasks.__dict__.items():
+            if callable(val) and name.startswith("_"):
```
Comment (singingwolfboy) [inline search/base.py]: Why did you add this condition?

Comment (noisecapella) [inline search/base.py]: I'm not sure it's necessary, I'll see if I can remove it.

Comment (singingwolfboy) [inline search/base.py]: I actually think that patching _fewer_ functions is better, if we can get away with it, so I think it's better to have the condition there. I just want to know why it was added, and what changed (if anything).

Comment (noisecapella) [inline search/base.py]: In `search.tasks` we are basically just importing functions from `search.indexing_api` with the same name. They are prefixed with `_` to prevent a name collision

Comment (singingwolfboy) [inline search/base.py]: Can you add a comment in the code to make that clearer?

Comment (noisecapella) [inline search/base.py]: Sure, just did
